### ISSUE TITLE
Feature Request: Search in extrafields

### ISSUE THREAD

**Issue by low5** [Dolibarr/dolibarr#3493]
Extrafields can´t be searched. This is a downside for handling customer specific requirements such as drawing numbers with products etc.

**Comment by jpfox** [comment]
It would be very cool :+1:

**Comment by mtahiue** [comment]
here is how i did  with dolibarr 4.0.3

caution !! : it is not a good pratice to hard code in the dolibarr core

```diff --git a/www/product/list.php b/www/product/list.php
index c95156b..1c3e608 100644
--- a/www/product/list.php
+++ b/www/product/list.php
@@ -118,6 +118,9 @@ $fieldstosearchall = array(
        'p.label'=>"ProductLabel",
        'p.description'=>"Description",
     "p.note"=>"Note",
+    "p.note"=>"Note",
+    "ef.sku"=>"SKU",
+    "ef.refadv"=>"Ref adv",
 );
```
note : adding and INDEX on thoses field in the mysql table  is required
```
ALTER TABLE `llx_product_extrafields` ADD INDEX(`refadv`);
ALTER TABLE `llx_product_extrafields` ADD INDEX(`sku`);
```
hope this will help a bit

**Comment by dolibarr95** [comment]
Could be very useful 👍

**Comment by jtraulle** [comment]
Seems to be still not implemented in V16.0.3

**Comment by eldy** [comment]
It seems that filter on extrafield is now possible for any fields with type text or number so i close the ticket
